src/mixer_plugin.c

```diff
@@ -151,13 +153,22 @@ static ssize_t mixer_plug_read_event(void *data, struct snd_ctl_event *ev, size_
     struct mixer_plugin *plugin = plug_data->plugin;
     eventfd_t evfd;
     ssize_t result = 0;
+    unsigned int i, read_cnt;
 
     result = plug_data->ops->read_event(plugin, ev, size);
 
     if (result > 0) {
-        plugin->event_cnt -=  result / sizeof(struct snd_ctl_event);
-        if (plugin->event_cnt == 0)
+        read_cnt = result / sizeof(struct snd_ctl_event);
+        pthread_mutex_lock(&plugin->mutex);
+        plugin->event_cnt -= read_cnt;
+        if (plugin->event_cnt < 0) {
```

Comment:
I'm curious why it is possible to be negative here. Is that possible to read 2 events but the second calls of mixer_plug_notifier_cb hasn't been called?

Reply:
previously in mixer_plug_notifier_cb, eventfd_write is called before event_cnt update, which may cause this negative count if read event is handled quickly before event_cnt update in mixer_plug_notifier_cb. then I think the negative check is added in android side (not upstreamed to github)

as current change is copied from android side fix hence we can see this check, but since we have mutex protection now so we can remove this check.